source/extensions/dynamic_modules/abi.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This is a pure single C header file that defines the ABI of the dynamic module
+// used by Envoy.
+//
+// This must not contain any dependencies besides standard library since it is not only used by
+// Envoy itself but also by dynamic module SDKs written in non-C++ languages.
+//
+// Currently, compatibility is only guaranteed by an exact version match between the Envoy
+// codebase and the dynamic module SDKs. In the future, after the ABI is stabilized, we will revisit
+// this restriction and hopefully provide a wider compatibility guarantee. Until then, we use the
+// simple symbol prefix "envoy_dynamic_module_" without any versioning at the symbol level.
+//
+// All ABI definitions, regardless of the area of usage, must be included in this single header
+// file. This is not only to ensure easy inclusion by any language bindings but also because the ABI
+// must be designed in a way that a single object file can be used at multiple extension points
+// (e.g., multiple TCP/HTTP filter chains) within the same Envoy process. The latter requirement
+// stems from the fact that some languages require exactly one shared library per process, such as
+// Go (https://github.com/golang/go/issues/65050). Additionally, it is important to reduce the
```

Comment:
I have a more general question: How does all of this relate to the existing Go plugin support? Isn't the end result of these efforts the same? Are we going to deprecate the other go plugin system for this one? It seems silly to support both.

Reply:
yeah I am thinking to deprecate the contrib/golang sinice all functionality provided there can be supported in this effort in Envoy code, and shouldn't be limited to Go at all. That's my current thinking. 

Anyways, I removed this comment lines are this sound not belong here, and made this PR focused on the exact version matching.